### DISCUSSION TITLE
Can I check if a transaction is currently started?

### DISCUSSION THREAD

**Discussion by timbze** [linq2db/linq2db#4485] [Q&A] [answered]
I have a method where I run more than 1 query...

There are times where it is run alone, and there are times where it's run as a part of a larger transaction.

Is there a way for me to run `BeginTransactionAsync()` only if a transaction has not been started already? Or in other words, can I check if a transaction is currently in progress?

**Comment by igor-tkachev** [comment, marked as answer]
Sure.

```c#
if (dataConnection.Transaction == null)
    dataConnection.Close();
```

**Reply by timbze** [reply]
Thank you!

Since I use an interface for my dbContext, I simply do it like this.

In DbContext (which implements DataConnection):
```c#
public bool TransactionStarted => Transaction is not null;
```

Then put that prop in my interface.
